ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -28,9 +29,10 @@ import (
 )
 
 const (
-	RayJobDefaultRequeueDuration    = 3 * time.Second
-	RayJobDefaultClusterSelectorKey = "ray.io/cluster"
-	PythonUnbufferedEnvVarName      = "PYTHONUNBUFFERED"
+	RayJobDefaultRequeueDuration        = 3 * time.Second
+	RayJobDefaultClusterSelectorKey     = "ray.io/cluster"
+	PythonUnbufferedEnvVarName          = "PYTHONUNBUFFERED"
+	DELETE_RAYJOB_CR_AFTER_JOB_FINISHES = "DELETE_RAYJOB_CR_AFTER_JOB_FINISHES"
```

Comment:
I suggest avoiding environment variables for this behavior. I think there are valid reasons why you want either behavior in a single cluster and we shouldn't force kuberay-operator to choose one or the other. 

How about a new field in RayJob like `spec.deleteAfterJobFinishes` or similar? Delete and shutdown after job finishes are two separate use-cases that I think warrant separate fields. They can be mutually exclusive so only one can be set at any time. `.spec.ttlSecondsAfterFinished` applies for both scenarios.

What do you think @MortalHappiness @kevin85421?

Reply:
I discussed with @kevin85421 few days ago. I also suggest that it may be better to change the RayJob CRD. However, he told me that he did't want the CRD be changed and this behavior can be implemented in cluster level. @kevin85421 What do you think?

Reply:
The current use case for this feature is to avoid having platform engineers periodically clean up submitter K8s Jobs. This feature is not related to Ray applications. That's why I prefer to set it at the KubeRay operator level. If there are use cases related to Ray applications, we can add a new field in CRD in the future to overwrite the behavior of the KubeRay operator level configuration.

I always try to avoid updating CRDs until there are clear use cases for doing so. It is hard to change, remove, or rename fields in CRDs.